lib/msf/core/web_services/json_rpc_exception_handling.rb

```diff
@@ -0,0 +1,34 @@
+require 'msf/core/rpc'
+
+module Msf::WebServices
+  module JsonRpcExceptionHandling
+
+    def self.registered(app)
+      app.set :show_exceptions => false
+
+      app.error { |err|
+        self.request.body.rewind
+        request = self.request.body.read
+
+        begin
+          request = JSON.parse(request, symbolize_names: true)
+        rescue JSON::ParserError
+          request = nil
```

Comment:
Since we're returning the request body in the error message maybe it would be better to _not_ return nil and instead keep it as it was sent to us, so we can more easily see any issues with the payload?

Reply:
Agreed, changing it to `inital_request` for clarities' sake

Reply:
We could return the entire request in the body of the message to allow users to discern what request caused the error, but the request `id` that is extracted from `self.request` allows users to do that anyway. And if we can't get the ID it's beause we couldn't get json from the initial request so what we'd be returning probably wouldn't be all that useful anyway.